src/Module/Maintenance.php

```diff
@@ -43,7 +44,17 @@ public static function content(array $parameters = [])
 		}
 
 		$exception = new HTTPException\ServiceUnavailableException($reason);
-		$exception->httpdesc = DI::l10n()->t('System down for maintenance');
-		throw $exception;
+
+		header($_SERVER["SERVER_PROTOCOL"] . ' ' . $exception->getCode() . ' ' . DI::l10n()->t('System down for maintenance'));
+
+		$tpl = Renderer::getMarkupTemplate('exception.tpl');
+
+		return Renderer::replaceMacros($tpl, [
+			'$title' => DI::l10n()->t('System down for maintenance'),
+			'$message' => DI::l10n()->t('This Friendica node is currently in maintenance mode, either automatically because it is self-updating or manually by the node administrator. This condition should be temporary, please come back in a few minutes.'),
+			'$thrown' => is_site_admin() ? $reason : '',
```

Comment:
The reason should be displayed all the time, not only for admins.

Reply:
The automatic reasons aren't meant for users. Manual ones may, but we don't have a way to differentiate both.

Reply:
The automatic reasons don't reveal any sensitive data. They are meant to be displayed to all users so that they see that the system is doing something.

Reply:
Ok, I'll remove the condition.